julia/helpers/DependabotHelper.jl/src/functions.jl

```diff
@@ -0,0 +1,1139 @@
+
+            # Get dependencies and add resolved versions
+            dependencies = []
+            for (dep_name, dep_uuid) in project_info.deps
+                dep_info = Dict{String,Any}(
+                    "name" => dep_name,
+                    "uuid" => string(dep_uuid)
+                )
+
+                # Add version constraint if available in compat
+                if haskey(project_info.compat, dep_name)
+                    compat_spec = project_info.compat[dep_name]
+                    # Extract the original constraint string
+                    constraint_str = if isa(compat_spec, Pkg.Types.Compat)
+                        compat_spec.str  # This should be the original constraint string
+                    else
+                        string(compat_spec)
+                    end
+                    dep_info["requirement"] = constraint_str
+                else
+                    dep_info["requirement"] = "*"
+                end
+
+                # Add resolved version from manifest if available
+                if haskey(manifest_versions, dep_name)
+                    dep_info["resolved_version"] = manifest_versions[dep_name]
+                end
+
+                push!(dependencies, dep_info)
+            end
+
+            # Get dev-dependencies (extras)
+            dev_dependencies = []
+            for (dep_name, dep_uuid) in project_info.extras
+                dev_dep_info = Dict{String,Any}(
+                    "name" => dep_name,
+                    "uuid" => string(dep_uuid),
+                    "requirement" => "*"
+                )
+                push!(dev_dependencies, dev_dep_info)
+            end
+
+            # Get Julia version requirement
+            julia_version = ""
+            if haskey(project_info.compat, "julia")
+                compat_spec = project_info.compat["julia"]
+                julia_version = if isa(compat_spec, Pkg.Types.Compat)
+                    compat_spec.str
+                else
+                    string(compat_spec)
+                end
+            end
+
+            # Get manifest information if available
+            manifest_info = Dict{String,Any}()
+            manifest_file_path = if !isnothing(manifest_path)
+                manifest_path
+            else
+                joinpath(project_dir, "Manifest.toml")
+            end
+
+            # No additional code needed here - everything is processed above
+
+            return Dict{String,Any}(
+                "name" => name,
+                "version" => version,
+                "uuid" => uuid,
+                "julia_version" => julia_version,
+                "dependencies" => dependencies,
+                "dev_dependencies" => dev_dependencies,
+                "manifest" => manifest_info,
+                "project_path" => ctx.env.project_file,
+                "manifest_path" => manifest_file_path
+            )
+        finally
+            # Restore original environment
```

Comment:
This functionality (a `try`-`finally`, in which the original environment is `Pkg.activate`d in the `finally` block) is used in a couple different places. Can you factor it out into a separate function, as such?

```julia
function with_active_project(f::Function)
    original_env = Base.active_project()
    result = try
        f()
    finally
        Pkg.activate(original_env)
    end
    return result
end
```